modules/sync/syncs/httpsynchronization.cpp

```diff
@@ -125,174 +116,131 @@ void HttpSynchronization::start() {
 
 void HttpSynchronization::cancel() {
     _shouldCancel = true;
-    reset();
-}
-
-void HttpSynchronization::clear() {
-    cancel();
-    // TODO: Remove all files from directory.
-}
-
-size_t HttpSynchronization::nSynchronizedBytes() {
-    return _nSynchronizedBytes;
-}
-
-size_t HttpSynchronization::nTotalBytes() {
-    return _nTotalBytes;
-}
-
-bool HttpSynchronization::nTotalBytesIsKnown() {
-    return _nTotalBytesKnown;
-}
-
-void HttpSynchronization::createSyncFile() {
-    const std::string& directoryName = directory();
-    const std::string& filepath = directoryName + ".ossync";
-
-    std::filesystem::create_directories(directoryName);
-
-    std::ofstream syncFile(filepath, std::ofstream::out);
-    syncFile << "Synchronized";
-    syncFile.close();
-}
-
-bool HttpSynchronization::hasSyncFile() {
-    const std::string& path = directory() + ".ossync";
-    return std::filesystem::is_regular_file(path);
+    _state = State::Unsynced;
 }
 
 bool HttpSynchronization::trySyncFromUrl(std::string listUrl) {
-    HttpRequest::RequestOptions opt = {};
-    opt.requestTimeoutSeconds = 0;
-
-    SyncHttpMemoryDownload fileListDownload(std::move(listUrl));
-    fileListDownload.onProgress([&c = _shouldCancel](HttpRequest::Progress) {
+    HttpMemoryDownload fileListDownload(std::move(listUrl));
+    fileListDownload.onProgress([&c = _shouldCancel](int64_t, std::optional<int64_t>) {
```

Comment:
Ooh, nice. Did not know it was possible to capture member variables like this, but makes sense :)

Reply:
If you wanna capture a std::unique_ptr, you need to do it like this:
```
std::unique_ptr<Foo> foo = ...;
auto fn = [f = std::move(foo)]() {  do(f); };
```